### PR TITLE
Remove hook advanced stats2

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
This is a second try of https://github.com/ankidroid/Anki-Android/pull/6274#pullrequestreview-417776941 , following the ideas of @david-allison-1. I make another PR so that if he changes his mind, we still have easy access to current version.

I personally agree with his suggestion and now that I've seen this version of the code, I prefer it

Obsoletes #6274

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
"Codacy was unable to analyse your pull request."
I've never seen that

File: AnkiDroid/src/main/java/com/ichi2/libanki/stats/AdvancedStatistics.java [hunk 1]
```diff
@@ -172,9 +158,13 @@ public static void uninstall(Hooks h) {
      * @param dids Deck id's
      * @return @see #metaInfo
      */
-    private StatsMetaInfo calculateDueAsMetaInfo(StatsMetaInfo metaInfo, Stats.AxisType type, Context context, String dids) {
+    public StatsMetaInfo calculateDueAsMetaInfo(StatsMetaInfo metaInfo, Stats.AxisType type, Context context, String dids) {
 
+        if (!PreferenceManager.getDefaultSharedPreferences(context.getApplicationContext()).getBoolean("advanced_statistics_enabled", false)) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/stats/AdvancedStatistics.java:163]:
```suggestion
        if (!AnkiDroidApp.getSharedPrefs(context).getBoolean("advanced_statistics_enabled", false)) {
```

**Comment by david-allison-1** [comment]
Travis failed:

```
C:\Users\travis\build\ankidroid\Anki-Android\AnkiDroid\src\main\java\com\ichi2\libanki\stats\AdvancedStatistics.java:163: error: cannot find symbol
        if (!AnkiDroidApp.getSharedPrefs(context).getBoolean("advanced_statistics_enabled", false)) {
             ^
  symbol:   variable AnkiDroidApp
  location: class AdvancedStatistics
```

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/libanki/stats/AdvancedStatistics.java  16
- AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.java  41
         

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java  -2
         

Clones added
============
- AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5612153&bid=18385435)

**Comment by Arthur-Milchior** [comment]
I was expecting to do it. No worries.
To be honest, I expect even worse when we'll want to merge PR related to making anki faster, since many deal with adding background task and changing the scheduler

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/libanki/stats/AdvancedStatistics.java  16
- AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.java  41
         

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java  -2
         

Clones added
============
- AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5612153&bid=18385435)

**Pull request merged**